apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,286 @@
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <task_manager/task_manager.h>
+#include "tc_common.h"
+
+#define TM_SAMPLE_NAME "tm_sample"
+#define TM_NOPERM_NAME "tm_noperm"
+#define TM_SAMPLE_MSG  "test_msg"
+#define TM_INVALID_HANDLE      -2
+#define TM_UNREGISTERED_HANDLE 99
+#define TM_INVALID_TIMEOUT     -2
+#define TM_INVALID_PERMISSION  -2
+#define TM_INVALID_UPPER_RANGE_PERMISSION 3
+#define TM_INVALID_UPPER_RANGE_HANDLE 33
+
+static int tm_sample_handle;
+static int tm_noperm_handle;
+static bool flag;
+static task_info_t *sample_info;
+static task_info_t *second_sample_info;
+
+static void test_handler(void *info)
+{
+	flag = !strncmp(info, TM_SAMPLE_MSG, strlen(TM_SAMPLE_MSG));
+}
+
+int itc_sample_main(int argc, char *argv[]) {
+	int ret;
+
+	tm_noperm_handle = task_manager_register(TM_NOPERM_NAME,
+	TM_TASK_PERMISSION_DEDICATE, 100);
+
+	ret =task_manager_set_unicast_cb(test_handler);
+	if (ret != OK) {
+		printf("ERROR : fail to set handler\n");
+	}
+	while (1)
+		;	// This will be dead at itc_task_manager_terminate_p
+}
+
+/**
+ * @testcase         itc_task_manager_clean_info_twice
+ * @brief            Get task_manager_getinfo_with_handle twice
+ * @scenario         Get task_manager_getinfo_with_handle twice
+ * @apicovered       task_manager_getinfo_with_handle
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_clean_info_twice(void) {
+	sample_info = (task_info_t *) task_manager_getinfo_with_handle(
+			tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t * )&sample_info, NULL);
+
+	task_manager_clean_info(&sample_info);
+	TC_ASSERT_NEQ("task_manager_clean_info", (task_info_t * )&sample_info, NULL);
+
+	TC_SUCCESS_RESULT()
+	;
+}
+
+
+/**
+ * @testcase         itc_task_manager_register_n
+ * @brief            Register task manager with invalid parameter
+ * @scenario         Register task manager with invalid parameter
+ * @apicovered       task_manager_register
+ * @precondition     NA
+ * @postcondition    NA
+ */
+
+static void itc_task_manager_register_n(void) {
+	int ret;
+	ret = task_manager_register(NULL, TM_TASK_PERMISSION_DEDICATE,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_INVALID_PERMISSION,
+	TM_NO_RESPONSE);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP,
+	TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME,
+	TM_INVALID_UPPER_RANGE_PERMISSION, TM_INVALID_TIMEOUT);
+	TC_ASSERT_EQ("task_manager_register_n", ret, TM_INVALID_PARAM);
+
+	ret = task_manager_register(TM_SAMPLE_NAME, TM_TASK_PERMISSION_GROUP,
```

Comment:
I think this ret cannot be TM_INVALID_PARAM

Reply:
Since I am giving INVALID PERMISSION as argument to task_manager_register() api, it is supposed to return TM_INVALID_PARAM. I also executed this tc and is working fine.